test/test-systemd-tmpfiles.py

```diff
@@ -31,6 +31,14 @@
 exe_with_args = sys.argv[1:]
 temp_dir = tempfile.TemporaryDirectory(prefix='test-systemd-tmpfiles.')
 
+# If /tmp isn't owned by either 'root' or the current user
+# systemd-tmpfiles will exit with "Detected unsafe path transition"
+# breaking this test
+tmpowner = os.stat("/tmp").st_uid
```

Comment:
i am sorry, but what? why does toolbox do that? that sounds like a really bad idea! a /tmp/ that isn't mapped properly? there are myriads of tmp file apis in libc and elswhere that assume /tmp just works.

This doesn't really make much sense?

Reply:
I'm not sure what the problem is that you're seeing, as (AFAICT) everything is working fine with `/tmp` - it is still world writable with the sticky bit set.   In fact I'm thinking toolbox isn't actually doing anything here, rather it is this way by inaction. Toolbox has passed through the host `/tmp` directory, and because host `UID=0` has **not** got a user namespace mapping set, the `/tmp` dir ownership gets mapped to `UID=65536 (nobody)` by the kernel usernamespace logic.

Reply:
For test cases like these we shouldn't choke on slightly weird setups - production code is different ofc. But we already have a few workarounds for other CI systems/runners, so if this is working as intended then it's fine to add it I think. Other comment needs to be fixed though.

Reply:
hmm, i had assumed that we aren't able to write to dirs of unmapped users.

This is all very fishy, it appears to be a bug in toolbox to me...

Reply:
> hmm, i had assumed that we aren't able to write to dirs of unmapped users.

In general that would be the case, but /tmp is unusual in that it is is intentionally world writable.

> This is all very fishy, it appears to be a bug in toolbox to me...

I don't think they're doing anything unusual here. They're calling podman create and telling it  `--userns keep-id` which sets up a userns mapping for the current unprivileged user ID, everything else gets squashed to `nobody`, and then they're also telling podman to map `/tmp` from the host to the container.